src/SFA.DAS.QnA.Application/Commands/SetPageAnswers/SetAnswersBase.cs

```diff
@@ -196,28 +196,23 @@ protected Next GetNextActionForPage(Guid sectionId, string pageId)
 
         private static bool CheckAllConditionsSatisfied(Condition condition, string questionTag)
         {
-            var allConditionsSatisfied = true;
-            
-
-            if (string.IsNullOrEmpty(questionTag))
+            if ((string.IsNullOrEmpty(condition.Contains)) && ((!string.IsNullOrEmpty(condition.MustEqual) && questionTag != condition.MustEqual) 
+                || (string.IsNullOrEmpty(condition.MustEqual) && !string.IsNullOrEmpty(questionTag))))
             {
-                allConditionsSatisfied = false;
+                return false;
             }
-            else if (!string.IsNullOrEmpty(condition.MustEqual) && questionTag != condition.MustEqual)
-            {
-                allConditionsSatisfied = false;
-            }
-            else if (!string.IsNullOrEmpty(condition.Contains))
+            
+            if (!string.IsNullOrEmpty(condition.Contains))
             {
                 var listOfAnswers = questionTag
                     .Split(",", StringSplitOptions.RemoveEmptyEntries);
                 if (!listOfAnswers.Contains(condition.Contains))
                 {
-                    allConditionsSatisfied = false;
+                    return false;
                 }
             }
 
-            return allConditionsSatisfied;
+            return true;
```

Comment:
bit pendant but could it follow the previous structure of if , else if , else if, return allConditionsSatisfied

Easier to comprehend than looking at the multiple returns

Reply:
Done